File: web/src/services/mediation.service.ts
import { apiClient } from '../config/api';
import type { MediationListResponse, InterestExpression } from '../types/mediation';

export interface ExpressInterestPayload {
  propertyId: string;
  matchId?: string;
  message?: string;
  interestType?: string;
  priority?: string;
}

export interface MediationFilter {
  connectionStatus?: string;
  includeAll?: boolean;
  propertyId?: string;
  buyerId?: string;
  sellerId?: string;
  page?: number;
  limit?: number;
}

export const mediationService = {
  async expressInterest(payload: ExpressInterestPayload): Promise<InterestExpression> {
    const response = await apiClient.post('/mediation/interest', payload);
    return response.data as InterestExpression;
  },
  async getMyInterests(filter: MediationFilter): Promise<MediationListResponse> {
    const response = await apiClient.get('/mediation/my-interests', { params: filter });
    return response.data as MediationListResponse;
  },
  async getPropertyInterests(filter: MediationFilter): Promise<MediationListResponse> {
    const response = await apiClient.get('/mediation/property-interests', { params: filter });
    return response.data as MediationListResponse;
  },
  async getPendingInterests(filter: MediationFilter): Promise<MediationListResponse> {
    const response = await apiClient.get('/mediation/pending', { params: filter });
    return response.data as MediationListResponse;
  },
  async reviewBuyerSeriousness(payload: Record<string, any>): Promise<InterestExpression> {
    const response = await apiClient.post('/mediation/review/buyer-seriousness', payload);
    return response.data as InterestExpression;
  },
  async reviewSellerWillingness(payload: Record<string, any>): Promise<InterestExpression> {
    const response = await apiClient.post('/mediation/review/seller-willingness', payload);
    return response.data as InterestExpression;
  },
  async approveConnection(payload: Record<string, any>): Promise<InterestExpression> {
    const response = await apiClient.post('/mediation/approve-connection', payload);
    return response.data as InterestExpression;
  },
  async rejectConnection(id: string, reason?: string): Promise<InterestExpression> {
    const response = await apiClient.post(`/mediation/reject-connection/${id}`, { reason });
    return response.data as InterestExpression;
  },
};

File: mobile/lib/services/mediation_service.dart
import '../config/api_client.dart';
import '../models/mediation.dart';

class MediationService {
  final _apiClient = ApiClient();

  Future<InterestExpression> expressInterest(Map<String, dynamic> payload) async {
    final response = await _apiClient.dio.post('/mediation/interest', data: payload);
    return InterestExpression.fromJson(Map<String, dynamic>.from(response.data));
  }

  Future<MediationListResponse> getMyInterests({String? status, String? propertyId, int page = 1, int limit = 20}) async {
    final response = await _apiClient.dio.get('/mediation/my-interests', queryParameters: {
      if (status != null && status.isNotEmpty) 'connectionStatus': status,
      if (propertyId != null && propertyId.isNotEmpty) 'propertyId': propertyId,
      'page': page,
      'limit': limit,
    });
    return MediationListResponse.fromJson(Map<String, dynamic>.from(response.data));
  }

  Future<MediationListResponse> getPropertyInterests({String? status, int page = 1, int limit = 20}) async {
    final response = await _apiClient.dio.get('/mediation/property-interests', queryParameters: {
      if (status != null && status.isNotEmpty) 'connectionStatus': status,
      'page': page,
      'limit': limit,
    });
    return MediationListResponse.fromJson(Map<String, dynamic>.from(response.data));
  }

  Future<MediationListResponse> getPendingInterests({String? status, int page = 1, int limit = 20}) async {
    final response = await _apiClient.dio.get('/mediation/pending', queryParameters: {
      if (status != null && status.isNotEmpty) 'connectionStatus': status,
      if (status == null || status.isEmpty) 'includeAll': true,
      'page': page,
      'limit': limit,
    });
    return MediationListResponse.fromJson(Map<String, dynamic>.from(response.data));
  }

  Future<InterestExpression> reviewBuyerSeriousness(Map<String, dynamic> payload) async {
    final response = await _apiClient.dio.post('/mediation/review/buyer-seriousness', data: payload);
    return InterestExpression.fromJson(Map<String, dynamic>.from(response.data));
  }

  Future<InterestExpression> reviewSellerWillingness(Map<String, dynamic> payload) async {
    final response = await _apiClient.dio.post('/mediation/review/seller-willingness', data: payload);
    return InterestExpression.fromJson(Map<String, dynamic>.from(response.data));
  }

  Future<InterestExpression> approveConnection(Map<String, dynamic> payload) async {
    final response = await _apiClient.dio.post('/mediation/approve-connection', data: payload);
    return InterestExpression.fromJson(Map<String, dynamic>.from(response.data));
  }

  Future<InterestExpression> rejectConnection(String id, {String? reason}) async {
    final response = await _apiClient.dio.post('/mediation/reject-connection/$id', data: {'reason': reason});
    return InterestExpression.fromJson(Map<String, dynamic>.from(response.data));
  }
}

